### ISSUE TITLE
[BUG] Android popup. Layout breaks if data is populated after the popup is shown.

### ISSUE THREAD

**Issue by Hooterr** [CommunityToolkit/Maui#1423]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Did you read the "Reporting a bug" section on Contributing file?

- [x] I have read the "Reporting a bug" section on Contributing file: https://github.com/CommunityToolkit/Maui/blob/main/CONTRIBUTING.md#reporting-a-bug

### Current Behavior

Populating labels after the popup is shown causes layout issues on Android. This is a huge issue if you need to use any sort of binding inside the popup.

When the labels are populated immediately

![Screenshot_1695383002](https://github.com/CommunityToolkit/Maui/assets/28758721/5fa960ac-272f-4938-a22e-9f8421f06b4e)

When the labels are populated a split second later, by a binding for isntance.
![Screenshot_1695383089](https://github.com/CommunityToolkit/Maui/assets/28758721/05ccc9d6-0ec6-4c89-9a54-2bb7c20d7b43)

As you can see the longer labels don't split into multiple lines correctly, or the single line label doesn't respect `HorizontalTextAlignment`. I have tried couple of layouts including `Grid` and `StackLayout` or `ScrollView` but no combination fixes the issue.

### Expected Behavior

Changing / Setting text values for labels or other text based controls shouldn't break layouts inside the popup.

### Steps To Reproduce

1. Show a popup
2. After it's visible set `Text` values for the labels

### Link to public reproduction project repository

https://github.com/Hooterr/PopupLayoutIssue

### Environment

```markdown
- .NET MAUI CommunityToolkit: 5.3.0
- OS: Macos
- .NET MAUI: 7.0.92
```

**Comment by cat0363** [comment]
This is caused by the following process added to redraw the Popup when the screen is rotated.
I'm writing it here for my personal notes.

[src\CommunityToolkit.Maui.Core\Handlers\Popup\PopUpHandler.android.cs]

    void OnLayoutChange(object? sender, EventArgs e)
    {
        if (VirtualView?.Handler?.PlatformView is Dialog dialog && Container is not null)
        {
            PopupExtensions.SetSize(dialog, VirtualView, Container);
        }
    }

Abolishing the above call will resolve this issue, but another problem will occur because redrawing 
will not occur when the screen is rotated.

**Comment by cat0363** [comment]
The PR #1361 I created should resolve this issue.

https://github.com/CommunityToolkit/Maui/assets/125236133/c2a44c37-edd6-49d9-9a3c-0d8b541633b6

I thought this was an unresolved issue and was investigating it separately, 
but it is scheduled to be resolved in PR #1361.There was no problem with screen rotation, 
which I was concerned about.

**Comment by cat0363** [comment]
It looks like PR #1361 needs further fixes.
If you look closely at the video, the position of the label for Test123 is different.

**Comment by Hooterr** [comment]
As a workaround just put `WidthRequest="300"` (or whatever your number is) on the top most layout in your popup.
Probably will break when the device is rotated, but at least you can use it now.

**Comment by brminnick** [comment]
@cat0363 Just to confirm - does PR #1361 now fix this Issue? Or will we need a separate PR to fix this Issue?

**Comment by cat0363** [comment]
@brminnick , I resolved this issue with PR #1361.
There is no need to create a new PR. FYI, I wrote it in the post below.
https://github.com/CommunityToolkit/Maui/pull/1361#issuecomment-1736908832

Thank you for your review of PR #1361.

**Comment by brminnick** [comment]
Fixed in https://github.com/CommunityToolkit/Maui/issues/1423
